### PR TITLE
HELP-20718: clarify format_from_uri and provide means to control per…

### PR DESCRIPTION
**Pull request by k-anderson**
…-account

### CODE REVIEW

**Comment by fenollp** [comment]
I'm scared by this

File: applications/stepswitch/src/stepswitch_util.erl [hunk 1]
```diff
@@ -243,10 +241,10 @@ set_endpoint_caller_id(Endpoint, Name, Number) ->
 
 -spec format_endpoint(wh_json:object(), api_binary(), filter_fun(), api_binary(), wh_json:object(), ne_binary()) ->
                              wh_json:object().
-format_endpoint(Endpoint, Number, FilterFun, DefaultRealm, SIPHeaders, AccountId) ->
+format_endpoint(Endpoint, Number, FilterFun, OffnetReq, SIPHeaders, AccountId) ->
```
Comment (lazedo) [inline applications/stepswitch/src/stepswitch_util.erl]: the change of `DefaultRealm` to `OffnetReq` needs a
change in spec from `api_binary()` to `wapi_offnet_resource:req()`

File: applications/stepswitch/src/stepswitch_util.erl [hunk 2]
```diff
@@ -283,14 +281,15 @@ maybe_add_sip_headers(Endpoint, SIPHeaders) ->
 
 -spec maybe_endpoint_format_from(wh_json:object(), ne_binary(), api_binary()) ->
                                         wh_json:object().
-maybe_endpoint_format_from(Endpoint, Number, DefaultRealm) ->
+maybe_endpoint_format_from(Endpoint, Number, OffnetReq) ->
```
Comment (lazedo) [inline applications/stepswitch/src/stepswitch_util.erl]: the change of `DefaultRealm` to `OffnetReq` needs a
change in spec from `api_binary()` to `wapi_offnet_resource:req()`

File: applications/stepswitch/src/stepswitch_util.erl [hunk 3]
```diff
@@ -300,10 +299,11 @@ maybe_endpoint_format_from(Endpoint, Number, DefaultRealm) ->
 
 -spec endpoint_format_from(wh_json:object(), ne_binary(), api_binary(), wh_json:object()) ->
                                   wh_json:object().
-endpoint_format_from(Endpoint, Number, DefaultRealm, CCVs) ->
-    case wh_json:get_value(<<"From-URI-Realm">>, CCVs, DefaultRealm) of
+endpoint_format_from(Endpoint, Number, OffnetReq, CCVs) ->
```
Comment (lazedo) [inline applications/stepswitch/src/stepswitch_util.erl]: the change of `DefaultRealm` to `OffnetReq` needs a
change in spec from `api_binary()` to `wapi_offnet_resource:req()`

**Pull request merged**